src/main/java/net/dries007/tfc/util/agriculture/Crop.java

```diff
@@ -25,58 +25,116 @@
 
 public enum Crop implements ICrop
 {
-    BARLEY(Food.BARLEY, 5f, 35f, 100f, 400f, 8, 0.5f, SIMPLE),
-    MAIZE(Food.MAIZE, 5f, 35f, 100f, 400f, 6, 0.5f, SIMPLE),
-    OAT(Food.OAT, 5f, 35f, 100f, 400f, 8, 0.5f, SIMPLE),
-    RICE(Food.RICE, 5f, 35f, 100f, 400f, 8, 0.5f, SIMPLE),
-    RYE(Food.RYE, 5f, 35f, 100f, 400f, 8, 0.5f, SIMPLE),
-    WHEAT(Food.WHEAT, 5f, 35f, 100f, 400f, 8, 0.5f, SIMPLE),
-    BEET(Food.BEET, 5f, 35f, 100f, 400f, 7, 0.5f, SIMPLE),
-    CABBAGE(Food.CABBAGE, 5f, 35f, 100f, 400f, 6, 0.5f, SIMPLE),
-    CARROT(Food.CARROT, 5f, 35f, 100f, 400f, 5, 0.5f, SIMPLE),
-    GARLIC(Food.GARLIC, 5f, 35f, 100f, 400f, 5, 0.5f, SIMPLE),
-    GREEN_BEAN(Food.GREEN_BEAN, 5f, 35f, 100f, 400f, 7, 0.5f, SIMPLE),
-    ONION(Food.ONION, 5f, 35f, 100f, 400f, 7, 0.5f, SIMPLE),
-    POTATO(Food.POTATO, 5f, 35f, 100f, 400f, 7, 0.5f, SIMPLE),
-    SOYBEAN(Food.SOYBEAN, 5f, 35f, 100f, 400f, 7, 0.5f, SIMPLE),
-    SQUASH(Food.SQUASH, 5f, 35f, 100f, 400f, 8, 0.5f, SPREADING),
-    SUGARCANE(Food.SUGARCANE, 5f, 35f, 100f, 400f, 8, 0.5f, SIMPLE),
-    RED_BELL_PEPPER(() -> new ItemStack(ItemFoodTFC.get(Food.RED_BELL_PEPPER)), () -> new ItemStack(ItemFoodTFC.get(Food.GREEN_BELL_PEPPER)), 5f, 35f, 100f, 400f, 7, 0.5f, SIMPLE),
-    TOMATO(Food.TOMATO, 5f, 35f, 100f, 400f, 8, 0.5f, SIMPLE),
-    YELLOW_BELL_PEPPER(() -> new ItemStack(ItemFoodTFC.get(Food.YELLOW_BELL_PEPPER)), () -> new ItemStack(ItemFoodTFC.get(Food.GREEN_BELL_PEPPER)), 5f, 35f, 100f, 400f, 7, 0.5f, SIMPLE),
-    JUTE(() -> new ItemStack(ItemsTFC.JUTE), () -> ItemStack.EMPTY, 5f, 35f, 100f, 400f, 6, 0.5f, SIMPLE),
-    PUMPKIN(() -> new ItemStack(Blocks.PUMPKIN), () -> ItemStack.EMPTY, 5f, 35f, 100f, 400f, 8, 0.5f, SPREADING),
-    MELON(() -> new ItemStack(Blocks.MELON_BLOCK), () -> ItemStack.EMPTY, 5f, 35f, 100f, 400f, 8, 0.5f, SPREADING);
-
-    public static final PropertyInteger STAGE_8 = PropertyInteger.create("stage", 0, 7);
-    public static final PropertyInteger STAGE_7 = PropertyInteger.create("stage", 0, 6);
-    public static final PropertyInteger STAGE_6 = PropertyInteger.create("stage", 0, 5);
-    public static final PropertyInteger STAGE_5 = PropertyInteger.create("stage", 0, 4);
+    // todo: unique rain tolerances for each crop
+    BARLEY(Food.BARLEY,
```

Comment:
Everything here is automatically generated via spreadsheet. This means:
 - No extra newlines between arguments, it's ugly
 - Any values changed here should be changed in the spreadsheet first, as it's much easier to peruse that for balance changes than raw source code.

Reply:
values have been changed in the spreadsheet and the code now represents those changes.